code/game/machinery/newscaster.dm

```diff
@@ -1064,8 +1066,13 @@ var/list/obj/machinery/newscaster/allCasters = list() //Global list that will co
 	var/scribble=""
 	var/scribble_page = null
 
-/obj/item/weapon/newspaper/attack_self(mob/user as mob)
+/obj/item/weapon/newspaper/attack_self(var/mob/user)
```

Comment:
sanity

Reply:
can the proc even get called without a user?

Reply:
I'm referring to the sanity we were talking about before, make sure the user isn't incapacitated in case they are for example dead or unconscious while buckled into a chair and holding the newspaper.

This is absolutely possible and a lack of sanity in much oldcode lets you do a lot of actions while dead, assuming you died buckled into a chair or glued to your hand. Try it for yourself in testing.

Reply:
In that case shouldn't we add the checks to `/obj/item/proc/attack_self(mob/user)` ? and have children call ..()

Reply:
If you want to refactor all attack_self in the code, certainly be my guest

Reply:
aight fair enough